### PR TITLE
Adds support for styling nixvim 

### LINKED ISSUE #153: Add NixVim support
Add suport for [NixVim: A Neovim configuration system for nix](https://github.com/nix-community/nixvim).

Currently, enabling both `programs.neovim.enable = true;` and `programs.nixvim.enable = true;` causes the build to fail because both modules are trying to write to the same neovim configuration file. However, disabling `programs.neovim.enable = true;` does not extend `programs.nixvim`'s configuration with the colorscheme declaration, as it is not supported yet.

### PR DESCRIPTION
**Pull request by willemml**
Adds the nixvim target, with two additional options controlling transparency:
```nix
{
    stylix.target.nixvim.transparent_bg = {
        # Enables following terminal transparency in the main window area
        main = true;
        # Enables following terminal transparency in the sign column
        sign_column = true;
    };
}
```

This close #153 and supersedes #178. Thanks to @trueNAHO for the initial work.

### CODE REVIEW

**Review by kdb424 — APPROVED** [review]
Workin great here.

File: modules/nixvim/nixvim.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+              "highlight NonText ctermbg=none"
+            ])
+            ++ (lib.optionals transparency_cfg.sign_column [
+              "highlight SignColumn guibg=none"
+              "highlight SignColumn ctermbg=none"
+            ]));
+        in ''
+          vim.cmd([[
+          ${vim_cmds}
+          ]])
```
Comment (trueNAHO) [inline modules/nixvim/nixvim.nix:59]: Is there a way to set these options with native Lua code for improved performance?

Maybe this gets the job done: https://nix-community.github.io/nixvim/NeovimOptions/highlight/index.html

**Review by trueNAHO — CHANGES_REQUESTED** [review]

File: modules/nixvim/nixvim.nix (hunk 1 above)
Comment (willemml) [reply modules/nixvim/nixvim.nix:59]: I tried that and it did not seem to work for me, it is possible that I was using it wrong though.

Comment (willemml) [reply modules/nixvim/nixvim.nix]: Turns out I was using them wrong before. I got it working.

**Review by trueNAHO — APPROVED** [review]

**Comment by trueNAHO** [comment]
@willemml, could you sync [your branch](https://github.com/willemml/stylix/tree/feat/modules/nixvim-transparency) with `origin/master`?

**Comment by arjan-s** [comment]
So awesome when I find somebody already made what I'm looking for! Thanks! 👍🎉

**Comment by arjan-s** [comment]
Any chance this will be merged soon(ish)?

**Comment by arjan-s** [comment]
Perhaps you could add some more transparency options? I'd like to set more of them, like here: https://github.com/siph/nixvim-flake/blob/master/config/visuals/theme/transparent-background.nix

**Review by danth — APPROVED** [review]
Can confirm this doesn't cause any issues when nixvim is not installed. Thanks!

**Comment by Kernald** [comment]
When using `neovim`, Stylix also styles `luabar`, which doesn't seem to be the case with `nixvim`. Is this something that can be implemented easily?

**Pull request merged**